diff --git a/node_modules/css-tree/dist/csstree.js b/node_modules/css-tree/dist/csstree.js
index 9333729..ffca1a5 100644
--- a/node_modules/css-tree/dist/csstree.js
+++ b/node_modules/css-tree/dist/csstree.js
@@ -8178,7 +8178,6 @@
                 var enterRet = enter.call(context, node, item, list);
 
                 if (enterRet === breakWalk) {
-                    debugger;
                     return true;
                 }
 
