diff --git a/node_modules/nx/src/project-graph/build-project-graph.js b/node_modules/nx/src/project-graph/build-project-graph.js
index 9c6b723..1f63823 100644
--- a/node_modules/nx/src/project-graph/build-project-graph.js
+++ b/node_modules/nx/src/project-graph/build-project-graph.js
@@ -51,7 +51,10 @@ function buildProjectGraphUsingProjectFileMap(projectsConfigurations, projectFil
         const context = createContext(projectsConfigurations, nxJson, projectFileMap, filesToProcess);
         let projectGraph = yield buildProjectGraphUsingContext(nxJson, context, cachedFileData, projectGraphVersion, packageJsonDeps);
         const projectGraphCache = (0, nx_deps_cache_1.createCache)(nxJson, packageJsonDeps, projectGraph, rootTsConfig);
-        if (shouldWriteCache) {
+        if (
+            shouldWriteCache &&
+            JSON.stringify(cache) !== JSON.stringify(projectGraphCache)
+        ) {
             (0, nx_deps_cache_1.writeCache)(projectGraphCache);
         }
         projectGraph.allWorkspaceFiles = allWorkspaceFiles;
