diff --git a/node_modules/react-native-dynamic-fonts/DynamicFonts.m b/node_modules/react-native-dynamic-fonts/DynamicFonts.m
index 5c8388e..750d83d 100644
--- a/node_modules/react-native-dynamic-fonts/DynamicFonts.m
+++ b/node_modules/react-native-dynamic-fonts/DynamicFonts.m
@@ -17,7 +17,7 @@ @implementation DynamicFonts
 - (void) loadFontWithData:(CGDataProviderRef) fontDataProvider callback:(RCTResponseSenderBlock)callback
 {
   CGFontRef newFont = CGFontCreateWithDataProvider(fontDataProvider);
-  NSString *newFontName = (__bridge NSString *)CGFontCopyPostScriptName(newFont);
+  NSString *newFontName = (__bridge_transfer NSString *)CGFontCopyPostScriptName(newFont);
 
   UIFont* font = [UIFont fontWithName:newFontName size:16];
   if (font != nil) {
@@ -31,7 +31,7 @@ - (void) loadFontWithData:(CGDataProviderRef) fontDataProvider callback:(RCTResp
     CFStringRef errorDescription = CFErrorCopyDescription(error);
     NSLog(@"Failed to register font: %@", errorDescription);
 
-    callback(@[@"Failed to register font: %@", (__bridge NSString *)errorDescription]);
+    callback(@[@"Failed to register font: %@", (__bridge_transfer NSString *)errorDescription]);
 
     CFRelease(errorDescription);
     CGFontRelease(newFont);
