diff --git a/node_modules/react-native-cookies/android/build.gradle b/node_modules/react-native-cookies/android/build.gradle
index 149aa19..850fa59 100644
--- a/node_modules/react-native-cookies/android/build.gradle
+++ b/node_modules/react-native-cookies/android/build.gradle
@@ -30,5 +30,5 @@ repositories {
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:+'
 }
diff --git a/node_modules/react-native-cookies/ios/react-native-cookies.podspec b/node_modules/react-native-cookies/react-native-cookies.podspec
similarity index 84%
rename from node_modules/react-native-cookies/ios/react-native-cookies.podspec
rename to node_modules/react-native-cookies/react-native-cookies.podspec
index 2c5c166..6fd64b2 100644
--- a/node_modules/react-native-cookies/ios/react-native-cookies.podspec
+++ b/node_modules/react-native-cookies/react-native-cookies.podspec
@@ -1,6 +1,6 @@
 require 'json'
 
-package = JSON.parse(File.read(File.join(__dir__, '../package.json')))
+package = JSON.parse(File.read(File.join(__dir__, 'package.json')))
 
 Pod::Spec.new do |s|
   s.name                = package['name']
@@ -16,6 +16,6 @@ Pod::Spec.new do |s|
   s.requires_arc        = true
   s.platform            = :ios, "7.0"
   s.preserve_paths      = "*.framework"
-  s.source_files        = 'RNCookieManagerIOS/*.{h,m}'
+  s.source_files        = 'ios/RNCookieManagerIOS/*.{h,m}'
   s.dependency 'React'
 end
